graph/src/data/graphql/queryable_type.rs

```diff
@@ -0,0 +1,179 @@
+use crate::prelude::s;
+use crate::schema::{EntityType, Schema};
+use std::cmp::Ordering;
+use std::collections::BTreeMap;
+use std::hash::{Hash, Hasher};
+use std::mem;
+
+use super::{DocumentExt, ObjectTypeExt};
+
+#[derive(Copy, Clone, Debug)]
+pub enum QueryableType<'a> {
```

Comment:
I like the name `QueryableType` much better than `ObjectOrInterface`; as I understand it, adding `union` here is necessary because of the return type of the `sql` query element, right?

Reply:
Exactly! we want the user to select which type of response: JSON or CSV. This is also future-proof for possible new formats.